ignite/base/events_driven.py

```diff
@@ -282,12 +283,20 @@ def __init__(
     def __getattr__(self, attr: str) -> Any:
         evnts = None
         if attr in self._attr_to_events:
-            evnts = self._attr_to_events[attr]
+            if attr == "epoch":
+                evnts = self._attr_to_events["epoch"][0]  # Fetch EPOCH_STARTED
+            elif attr == "iteration":
+                evnts = self._attr_to_events["iteration"][2]  # Fetch ITERATION_STARTED
+            else:
+                evnts = self._attr_to_events[attr]
```

Comment:
as we talked in VC, this part of code is to remove

Reply:
I am not sure if that's can be done in the way we talked about, if we replaced `event_to_attr` arg with `attr_to_event` and if it's None, we will need to put manually so we won't avoid using `epoch` and `iteration` here. And also `_attr_to_event` is updated anyway within `update_mapping` method, so `iteration` will be fetched from `GET_BATCH_STARTED` also. So I'm thinking about either we override `__getattr__` in State or a more nice solution is to just override `update_mapping` in state to only map `ITERATION_STARTED` and `EPOCH_STARTED` in case of iteration/epoch
`update_mapping` in state maybe should be something like this, I tried it and it works fine
```py
 if event_to_attr:
            for k, v in event_to_attr.items():
                if v == 'epoch' and k != Events.EPOCH_STARTED:
                    attr_evnts = self._attr_to_events["epoch"]
                    continue
                if v == 'iteration' and k != Events.ITERATION_STARTED:
                    attr_evnts = self._attr_to_events["iteration"]
                    continue
                else:
                    attr_evnts = self._attr_to_events[v]
                if k not in attr_evnts:
                    attr_evnts.append(k)
```
What do you think?

Reply:
@KickItLikeShika again "iteration" and "epoch" are specific to Engine and State and shouldn't appear in any case in EventsDrivenState.

Reply:
Yes, that's why I said we will extend `update_mapping` to state and do that?

Reply:
This solution does not look better neither

Reply:
Can you please explain your point why it doesn't look nice, for me:
- Avoided mentioning `epoch`/`iteration` in `EventsDrivenState` and made this specific to `State`.
- Kept `EventsDrivenState` generic as it is with the same simple implementation.
- Not much duplication as `updating_mapping` is a 5 lines method.
I can push the updates so you can explain your concerns better. Thanks.

Reply:
Anyway, It doesn't work as expected, I will look for another solution, thanks.